src/Plugin/GeneratorProcessPluginDateBase.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Drupal\migrate_generator\Plugin;
+
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+use Drupal\datetime_range\Plugin\Field\FieldType\DateRangeItem;
+
+/**
+ * The base class for date migrate generator process plugins.
+ */
+abstract class GeneratorProcessPluginDateBase extends GeneratorProcessPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getBaseProcess($field_name, $source = NULL) {
+    $process = parent::getBaseProcess($field_name, $source);
+
+    $source_date_format = isset($this->options['date_format']) ? $this->options['date_format'] : 'd-m-Y H:i:s';
+    $field_type = $this->getFieldStorageDefinition()->getType();
+    $timestamp_types = ['created', 'changed'];
+    $datetime_types = ['datetime', 'daterange'];
```

Comment:
We should probably make these 2 into 2 different abstract classes
1 for Timestamp and another for Datetime.
Up to you if we do it right away or have it as follow-up

Reply:
not sure that we really need 2 abstract classes here, main difference is only in format of date